src/hnsw/node_cache.h

```diff
@@ -0,0 +1,27 @@
+#ifndef LDB_HNSW_NODE_CACHE_H
+#define LDB_HNSW_NODE_CACHE_H
+#include "postgres.h"
+
+#include <storage/block.h>
+#include <utils/hsearch.h>
+#include <utils/memutils.h>
+
+typedef int NodeCacheKey;
+typedef struct NodeCacheEntry
+{
+    NodeCacheKey key;
+    void        *value;
+} NodeCacheEntry;
+typedef struct NodeCache
+{
+    HTAB   *htab;
+    HASHCTL hctl;
+} NodeCache;
+
+NodeCache node_cache_create();
```

Comment:
Can we refactor these to have one `cache_` api?
It seems the only thing different between the two caches in this PR is the key and values size.
So, we can pass those in create call and have the rest of the arguments be `void*` in the data structure definition.

Maybe we could have a type-specific wrapper around the data structure for each instance (`node_cache` and `bln_cache`) to get some compiler help. Not sure if this is necessary or useful enough, though.

Reply:
Sure!